docs/AddingDoppl.md

```diff
@@ -0,0 +1,565 @@
+
+```groovy
+dopplConfig {
+  translatePattern {
+    include 'co/doppl/so/api/**'
+    include 'co/doppl/so/arch/**'
+    include 'co/doppl/so/RepositoryTest.java'
+  }
+
+  translatedPathPrefix 'co.doppl.so.api', 'SOAPI'
+  translatedPathPrefix 'co.doppl.so.arch', 'SOA'
+
+  testIdentifier {
+    include 'co/doppl/so/RepositoryTest.java'
+  }
+}
+```
+
+This tells Doppl to:
+
+- Convert the classes in our `co.doppl.so.api` and `co.doppl.so.arch` packages,
+along with the `co.doppl.so.RepositoryTest` class
+
+- When generating Objective-C files, use certain prefixes for the different
+Java package names
+
+- Mark the `co.doppl.so.RepositoryTest` class as being a unit test class
+that should be in the test output, not the production output, from the Doppl
+conversion process
+
+Some notes about this:
+
+- The `MainActivity` class is not converted, as that is our Android
+UI, and we will want to create a native iOS UI instead.
+
+- `translatePathPrefix` requires a match on the entire Java package name.
+For example, `translatedPathPrefix 'co.doppl', 'D'` would not match
+any classes, as there are no classes in this project *directly* in the
+`co.doppl` package.
+
+## Step #6: Define Doppl Dependencies
+
+The Doppl conversion process converts Java source code, not compiled Java
+bytecode or Dalvik bytecode. As a result, we need to provide Doppl with
+Doppl-specific alternatives to libraries that are used by the Java classes
+that are being converted.
+
+To that end, make the following modifications to the `build.gradle` file in the
+`app/` module.
+
+### Add Some Constants
+
+You should see a block of constants for some version numbers that are used
+for multiple artifacts:
+
+```groovy
+def supportVersion = "26.1.0"
+def archVer = "1.0.0-rc1"
+def retroVer = "2.3.0"
+```
+
+Add these two to that block:
+
+```groovy
+def dopplArchVer = "1.0.0.0-rc1"
+def dopplRetroVer = "2.3.0.7"
+```
+
+### Add Some doppl Statements
+
+Then, replace the existing `dependencies` closure with the following:
+
+```groovy
+dependencies {
+  implementation      "com.android.support.constraint:constraint-layout:1.0.2"
+  implementation      "com.android.support:recyclerview-v7:$supportVersion"
+  implementation("com.android.support:support-v4:$supportVersion") {
+    exclude group: 'com.android.support', module: 'support-media-compat'
+  } // for https://issuetracker.google.com/issues/64909326
+  implementation      "com.squareup.picasso:picasso:2.5.2"
+
+  implementation      "io.reactivex.rxjava2:rxjava:2.1.5"
+  doppl               "co.doppl.io.reactivex.rxjava2:rxjava:2.1.5.0"
+  implementation      "io.reactivex.rxjava2:rxandroid:2.0.1"
+  doppl               "co.doppl.io.reactivex.rxjava2:rxandroid:2.0.1.2"
+
+  implementation      "com.squareup.retrofit2:converter-gson:$retroVer"
+  implementation      "com.squareup.retrofit2:adapter-rxjava2:$retroVer"
+  doppl               "co.doppl.com.squareup.retrofit2.urlsession:retrofit:$dopplRetroVer"
```

Comment:
The fact that there is no `implementation` to match this `doppl` shows it's not always a 1-1 relationship for the dependencies in shared code. This is confusing because I'd expect to only have to include `converter-gson` and `adapter-rxjava2` doppl alternatives.

Reply:
Yup, and I have no explanation for that. If somebody in the know has details, I'd love to add 'em.

Reply:
I think that `implementation "com.squareup.retrofit2:retrofit:$retroVer"' should be added, if only just for clarity. It looks like we are getting by without it because the adapter and converter depend on it. I'm not sure whether the doppl version could be omitted similarly, but I think it makes sense to leave both in

Reply:
Done -- thanks!